### PR TITLE
Bump up C++ version, modify googletest tag in cmake

### PR DESCRIPTION
**Pull request by asmelko**
should fix issues #13 and #12

### CODE REVIEW

File: cmake/external_gtest.cmake [hunk 1]
```diff
@@ -1,11 +1,11 @@
-cmake_minimum_required(VERSION 2.8.2)
+cmake_minimum_required(VERSION 3.0.0)
 
 project(googletest-download NONE)
 
 include(ExternalProject)
 ExternalProject_Add(googletest
   GIT_REPOSITORY    https://github.com/google/googletest.git
-  GIT_TAG           master
+  GIT_TAG           main
```
Comment (exaexa) [inline cmake/external_gtest.cmake]: maybe it would be better to depend on actual fixed tag to prevent future surprises (api breakage etc)

Comment (exaexa) [reply cmake/external_gtest.cmake]: (I hope they do have version tags, right?)

Comment (asmelko) [reply cmake/external_gtest.cmake]: And give up on [Live at Head](https://abseil.io/about/philosophy#upgrade-support) philosophy? :(

Comment (asmelko) [reply cmake/external_gtest.cmake]: all right :D

**Pull request merged**